30412/Keresztes Beata/Assignment 1 extra/tasks.c

```diff
@@ -0,0 +1,124 @@
+//
+// Created by q on 3/2/2020.
+//
+
+#include "tasks.h"
+
+int findMaxNoOfWaves(int M, RebelT * countries) {
+
+    int Max=0;
+    for(int i=0;i<M;i++) {
+        int max=0;
+        WaveT * temp=countries[i].first;
+        while(temp) {
+            max++;
+            temp=temp->nextWave;
+        }
+        if(max>Max) {
+            Max=max;
+        }
+    }
+    return Max;
+}
+void resultOfAttack(FILE *fp, int M, RebelT * countries, int resistance) {
+
+    WaveT ** attacks = (WaveT **) malloc(M * sizeof(WaveT *));
+    for (int i = 0; i < M; i++) {
+        attacks[i] = (WaveT *)malloc(sizeof(WaveT));
+        attacks[i] = countries[i].first;
+    }
+
+    int maxNoOfWaves = findMaxNoOfWaves(M, countries);
+
+    int amountOfDamage = 0, finalBlowIndex = -1;
+    bool victory = false;
+
+    for (int k = 0; k < maxNoOfWaves && !victory; k++) {
+
+        for (int i = 0; i < M && !victory; i++) {
+            if (attacks[i] != NULL) {
+                amountOfDamage += attacks[i]->damage;
+                attacks[i] = attacks[i]->nextWave;
+            }
+            if (amountOfDamage >= resistance) {
+                finalBlowIndex = i;
+                victory = true;
+            }
+        }
+    }
+
+    if (victory == true) {
```

Comment:
you don't need `== true` or `== false` when having a boolean inside an if.
`if (victory)` or `if (!victory)` is enough

Reply:
Also, would much rather move these ending conditions check into a different function